### DISCUSSION TITLE
Clear actions via midi binding?

### DISCUSSION THREAD

**Discussion by Flavelius** [monocasual/giada#617] [Q&A]
Hi, i'm not a perfect musician so i often make mistakes and need to rerecord things. So far it seems i can only clear midi actions via mouse-menu interactions, which breaks the flow. Is there a setting i just missed or is it currently really not possible to bind this command to a key/midi message?

**Comment by gvnnz** [comment]
Hi @Flavelius , thanks for the idea and sorry for the delay. You are making a good point here, currently it's a mouse-only operation. 
Question: would you just delete _any_ recorded MIDI action in a channel?

**Reply by Flavelius** [reply]
I'm not sure i fully understand. If I want to rerecord something I want to clear everything that previously was part of the recording. So, mainly notes I guess, not sure if there's anything else recordable. But not the vst settings or similar configuration

**Reply by gvnnz** [reply]
The "clear actions" command we have now is pretty limited, as it just removes all actions from a channel. Maybe is not good enough in a live situation. 

I was thinking about the following scenario. You have a MIDI channel with some actions already recorded; you then start a new recording session and add more MIDI actions to that channel (overdub). What would you expect after a "clear actions" operation? All actions are deleted or only those actions recorded in the second recording session?

**Reply by Flavelius** [reply]
I never use overdub, as I always find it better to record to a new track, so I cannot comment on that with background, but I would naturally expect all notes to be cleared, not just the last overdub, as that could be covered with some kind of undo.
